docs/runtime-function-reference.md

```diff
@@ -0,0 +1,691 @@
+
+_Returns_
+
+(string) - The message which was written to the logs.
+
+_Example_
+
+```lua
+local message = ("%q - %q"):format("hello", "world")
+nk.logger_warn(message)
+```
+
+### notifications
+
+__notifications_send_id (new_notifications)__
+
+Send one or more in-app notifications to a user. Have a look at the section on [in-app notifications](in-app-notifications.md).
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| new_notifications | table | The Lua table array of notifications to send. |
+
+_Example_
+
+```lua
+local subject = "You've unlocked level 100!"
+local content = nk.json_encode({
+  reward_coins = 1000
+})
+local user_id = "4c2ae592-b2a7-445e-98ec-697694478b1c" -- who to send
+local code = 101
+
+local new_notifications = {
+  { Subject = subject, Content = content, UserId = user_id, Code = code, Persistent = true}
+}
+nk.notifications_send_id(new_notifications)
+```
+
+
+
+### register hooks
+
+__register_after (func, msgname)__
+
+Register a function with the server which will be executed after every message with the specified message name.
+
+This can be used to apply custom logic to standard features in the server. Similar to the `register_before` function but it will not block the execution pipeline. The logic will be executed in parallel to any response message sent back to a client.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| func | function | A function reference which will be executed on each `msgname` message. |
+| msgname | string | The specific message name to execute the `func` function after. |
+
+_Example_
+
+```lua
+local function my_func(context, payload)
+  -- run some code
+end
+nk.register_after(my_func, "TFriendAdd")
+```
+
+---
+
+__register_before (func, msgname)__
+
+Register a function with the server which will be executed before every message with the specified message name.
+
+For example `register_before(somefunc, "TFriendAdd")` will execute the function before the Friend Add message is executed by the server’s message pipeline. This can be used to apply custom conditions to standard features in the server.
+
+_Parameters_
+
+| Param | Type | Description |
+| ----- | ---- | ----------- |
+| func | function | A function reference which will be executed on each `msgname` message. |
+| msgname | string | The specific message name to execute the `func` function before. |
+
+!!! note
+    The `func` should pass the `payload` back as a return argument so the pipeline can continue to execute the standard logic.
+
+_Example_
+
+```lua
+local function my_func(context, payload)
+  -- run some code
+  return payload -- important!
+end
+nk.register_before(my_func, "TFriendAdd")
+```
+
+---
+
+__register_http (func, path)__
+
+Registers a HTTP endpoint within the server.
+
+!!! warning
+    This should not be used to implement custom client functions use `register_rpc` instead.
```

Comment:
Missing dash.
"This should not be used to implement custom client functions - use `register_rpc` instead."